src/main/java/au/com/southsky/jfreesane/SaneDevice.java

```diff
@@ -214,4 +217,41 @@ SaneSession getSession() {
   void invalidateOptions() {
     optionTitleMap = null;
   }
+
+  @Override
+  public int hashCode() {
+    int hash = 3;
+    hash = 37 * hash + Objects.hashCode(this.name);
+    hash = 37 * hash + Objects.hashCode(this.vendor);
+    hash = 37 * hash + Objects.hashCode(this.model);
+    hash = 37 * hash + Objects.hashCode(this.type);
+    return hash;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
Can you comment on the motivation for this change?

I don't think this is right. It allows two devices on different SANE servers to be considered equal. Better might be:

* if the devices' `session` fields not equal, the devices are not equal
* if the devices' `handle` fields are not equal, the devices are not equal
* if neither device has a `handle field`, the devices are equal if their names are equal (because this is how SANE refers to them before a handle is assigned)